integration-tests/test-cases/scenario-tests/src/test/java/io/cellery/integration/scenario/tests/EmployeePortalTestCase.java

```diff
@@ -94,4 +140,234 @@ public void cleanup() {
+        String url = "https://wso2-apim/client-registration/v0.14/register";
+        String usernameColonPassword = "alice" + ":" + "alice123";
+        String parameters = "{\"callbackUrl\": \"www.google.lk\", \"clientName\": \"rest_api_store\", \"owner\": " +
+                "\"alice\", \"grantType\": \"password refresh_token\", \"saasApp\": true}";
+        String authenticationKey = Base64.getEncoder().encodeToString(usernameColonPassword.getBytes(
+                StandardCharsets.UTF_8));
+        //Add headers
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Authorization", "Basic " + authenticationKey);
+        headers.put("Content-Type", "application/json");
+        String response = httpClient.sendPost(url, parameters, headers);
+        JsonObject responseJson = new JsonParser().parse(response).getAsJsonObject();
+        Assert.assertTrue(responseJson.isJsonObject());
+        // Get client id and client secret from the response
+        String clientId = responseJson.get("clientId").getAsString();
+        String clientSecret = responseJson.get("clientSecret").getAsString();
+        return clientId + ":" + clientSecret;
+    }
+
+    /**
+     * Get the id of hr api.
+     *
+     * @return hr api id
+     * @throws NoSuchAlgorithmException if sendGet method fails
+     * @throws IOException              if sendGet method fails
+     * @throws KeyManagementException   if sendGet method fails
+     */
+    private String getHrApiId() throws NoSuchAlgorithmException, IOException
+            , KeyManagementException {
+        String hrApiId = "";
+        // Get the list of APIs
+        String url = "https://wso2-apim/api/am/store/v0.14/apis";
+        //Add headers
+        Map<String, String> headers = new HashMap<>();
+        String apis = httpClient.sendGet(url, headers);
+        JsonObject responseJson = new JsonParser().parse(apis).getAsJsonObject();
+        JsonArray apiList = responseJson.get("list").getAsJsonArray();
+        for (int i = 0; i < apiList.size(); i++) {
+            JsonObject jsonItem = apiList.get(i).getAsJsonObject();
+            // Get the id of hr api
+            if (jsonItem.get("name").toString().equals("\"hr_inst_global_1_0_0_hr_api\"")) {
+                hrApiId = jsonItem.get("id").toString().replaceAll("^\"|\"$", "");
+            }
+        }
+        return hrApiId;
+    }
+
+    /**
+     * Get the oauth token required to access wso2-apim.
+     *
+     * @param clientIdColonClientSecret A concatenated string comprising client id and client secret delimited by a colon
+     * @return oauth token
+     * @throws NoSuchAlgorithmException if sendPost method fails
+     * @throws IOException              if sendPost method fails
+     * @throws KeyManagementException   if sendPost method fails
+     */
+    private String getOauthToken(String clientIdColonClientSecret) throws NoSuchAlgorithmException, IOException
```

Comment:
This method also could go to utils or base test case. Further pass the username, and password as params to the method when moving it. This is a potential method which could be useful for other methods as well.

Reply:
Merged two functions "getClientIdColonClientSecret" and "getOauthToken" and created a function "getAccessTokenForApiStore" and moved it to util package. This will be a generic function which takes username and password as parameters.